students/YingGuo/lession07/html_render.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+
+
+# This is the framework for the base class
+class Element():
+    tag_name = "html"
+    indent = "    "
+
+    def __init__(self, content=None, **kwargs):
+        if content is None:
+            self.content = []
+        if content is not None:
+            self.content = [content]
+        self.kwargs = kwargs 
+
+
+    def append(self, new_content):
+        self.content.append(new_content)
+
+    def write_html(self):
+        with open("output_html","w") as f:
+             f.write(self.content)
+             
+    #what does ind mean in the test_html? why it is needed?
+    def render(self, out_file, cur_ind=""):
+        if cur_ind:
+            out_file.write(cur_ind)
+        else:
+            pass
+
+        for content in self.content:
+            out_file.write("<{}>\n".format(self.tag_name))
+
+            if cur_ind:
+                out_file.write(cur_ind)
+            else:
+                pass
+            #
+            out_file.write(self.indent)
+            try:
+                content.render(out_file)
+            except AttributeError:
+                out_file.write(content)
+            out_file.write("\n")
+            
+            out_file.write("</{}>\n".format(self.tag_name))
+
+
+#a subclass of Element with tag body
+class Body(Element):
+    tag_name = "body"
+    indent = "    "*2
```

Comment:
This is not right... You shouldn't hard-code the indentation level. It should have been programmatically handled.